### PR TITLE
server: Add rangefeed metrics

### PR DESCRIPTION
**Pull request by miretskiy**
Add rangefeed related metrics to keep track of the number of actively running rangefeeds on the server.

Epic: CRDB-26372
Release note: None

### CODE REVIEW

File: pkg/server/node.go [hunk 1]
```diff
@@ -195,6 +195,18 @@ This metric is thus not an indicator of KV health.`,
 		Measurement: "Bytes",
 		Unit:        metric.Unit_BYTES,
 	}
+	metaActiveRangeFeed = metric.Metadata{
+		Name:        "stream.rangefeed.count",
```
Comment (erikgrinaker) [inline pkg/server/node.go:199]: How about an `rpc.streams.` prefix for consistency with other RPC metrics, and have both a counter and a gauge for each method?

We could also consider generating automatic metrics for all streaming methods, similarly to `rpc.method.%s.recv` for regular requests. But we can punt that as long as the metrics scheme is extensible.

File: pkg/server/node.go [hunk 2]
```diff
@@ -195,6 +195,30 @@ This metric is thus not an indicator of KV health.`,
 		Measurement: "Bytes",
 		Unit:        metric.Unit_BYTES,
 	}
+	metaActiveRangeFeed = metric.Metadata{
+		Name:        "rpc.streams.rangefeed.count",
+		Help:        `Number of currently running RangeFeed streams`,
+		Measurement: "Count",
+		Unit:        metric.Unit_BYTES,
```
Comment (erikgrinaker) [inline pkg/server/node.go:202]: `Unit_COUNT`

File: pkg/server/node.go [hunk 3]
```diff
@@ -195,6 +195,30 @@ This metric is thus not an indicator of KV health.`,
 		Measurement: "Bytes",
 		Unit:        metric.Unit_BYTES,
 	}
+	metaActiveRangeFeed = metric.Metadata{
+		Name:        "rpc.streams.rangefeed.count",
+		Help:        `Number of currently running RangeFeed streams`,
+		Measurement: "Count",
```
Comment (erikgrinaker) [inline pkg/server/node.go:201]: Streams

File: pkg/server/node.go [hunk 4]
```diff
@@ -195,6 +195,30 @@ This metric is thus not an indicator of KV health.`,
 		Measurement: "Bytes",
 		Unit:        metric.Unit_BYTES,
 	}
+	metaActiveRangeFeed = metric.Metadata{
+		Name:        "rpc.streams.rangefeed.count",
```
Comment (erikgrinaker) [inline pkg/server/node.go:199]: Count seems like a misnomer, since this is a gauge not a counter. Active? Current?

**Review by erikgrinaker — APPROVED** [review]

File: pkg/server/node.go (hunk 4 above)
Comment (miretskiy) [reply pkg/server/node.go:199]: Ack.

File: pkg/server/node.go (hunk 2 above)
Comment (miretskiy) [reply pkg/server/node.go:202]: done.

**Pull request merged**